pkg/prometheus/agent/statefulset.go

```diff
@@ -236,6 +255,24 @@ func makeStatefulSetSpec(
 		webConfigGenerator.Warn("web.config.file")
 	}
 
+	// Mount web config and web TLS credentials as volumes for config-reloader.
+	// We always mount the web config file for versions greater than 0.69.0.
+	if cfgReloaderVersion.GTE(semver.MustParse("0.69.0")) && cpf.WebConfigReloader != nil {
```

Comment:
Just wondering if we can abstract this code in a func and reuse between agent and server.

Reply:
Thank you for your review. But abstracting this code into a func will have too many parameters. Probably not worth it 🤔️

Reply:
I'm kinda Ok having some parameters on the func signature.
If we think it's to much we can always create an struct and send it to the func.

WDYT?